dolphin/task_manager/tasks/task.py

```diff
@@ -115,4 +128,31 @@ def sync(self):
             LOG.info("Syncing pools successful!!!")
 
     def remove(self):
+        LOG.info('Remove pools for storage id:{0}'.format(self.storage_id))
+        db.pool_delete(self.context, self.storage_id)
+
+
+class StorageVolumeTask(StorageResourceTask):
+    def sync(self):
         pass
+
+    def remove(self):
+        LOG.info('Remove volumes for storage id:{0}'.format(self.storage_id))
+        db.volume_delete(self.context, self.storage_id)
+
+
+class StorageResourceInMemoryTask(object):
```

Comment:
IMO, it's better to let task manager remove driver in cache.

Reply:
Done